src/wp-includes/blocks/navigation.php

```diff
@@ -1066,3 +1342,120 @@ function block_core_navigation_get_most_recently_published_navigation() {
 
 	return null;
 }
+
+/**
+ * Insert hooked blocks into a Navigation block.
+ *
+ * Given a Navigation block's inner blocks and its corresponding `wp_navigation` post object,
+ * this function inserts hooked blocks into it, and returns the serialized inner blocks in a
+ * mock Navigation block wrapper.
+ *
+ * If there are any hooked blocks that need to be inserted as the Navigation block's first or last
+ * children, the `wp_navigation` post's `_wp_ignored_hooked_blocks` meta is checked to see if any
+ * of those hooked blocks should be exempted from insertion.
+ *
+ * @param array   $inner_blocks Parsed inner blocks of a Navigation block.
+ * @param WP_Post $post         `wp_navigation` post object corresponding to the block.
```

Comment:
```suggestion
 * @param array        $inner_blocks Parsed inner blocks of a Navigation block.
 * @param WP_Post|null $post         Optional. `wp_navigation` post object corresponding to the block. Default null.
```

Reply:
cc @tjcafferkey
 
Can you please follow-up with a PR in Gutenberg to address the comments that are related to the changes here